mlflow/utils/autologging_utils.py

```diff
@@ -306,6 +310,97 @@ def batch_metrics_logger(run_id):
     batch_metrics_logger.flush()
 
 
+def _check_version_in_range(ver, min_ver, max_ver):
+    return LooseVersion(min_ver) <= LooseVersion(ver) <= LooseVersion(max_ver)
+
+
+def _load_version_file_as_dict():
+    version_file_path = resource_filename(__name__, "../ml-package-versions.yml")
+    with open(version_file_path) as f:
+        return yaml.load(f, Loader=yaml.SafeLoader)
+
+
+_module_version_info_dict = _load_version_file_as_dict()
+
+
+# A map FLAVOR_NAME -> list of dependent (module_name, key_in_version_file_json)
+_cross_tested_flavor_to_module_name_and_module_key = {
+    "fastai": [("fastai", "fastai-1.x")],
+    "gluon": [("mxnet", "gluon")],
+    "keras": [("keras", "keras")],
+    "lightgbm": [("lightgbm", "lightgbm")],
+    "statsmodels": [("statsmodels", "statsmodels")],
+    "tensorflow": [("tensorflow", "tensorflow")],
+    "xgboost": [("xgboost", "xgboost")],
+    "sklearn": [("sklearn", "sklearn")],
+    "pytorch": [("torch", "pytorch"), ("pytorch_lightning", "pytorch-lightning")],
+}
+
+
+def _get_min_max_version_and_pip_release(module_key):
+    min_version = _module_version_info_dict[module_key]["autologging"]["minimum"]
+    max_version = _module_version_info_dict[module_key]["autologging"]["maximum"]
+    pip_release = _module_version_info_dict[module_key]["package_info"]["pip_release"]
+    return min_version, max_version, pip_release
+
+
+def is_autologging_integration_supported(flavor_name):
+    def _check_supported(module_name, module_key):
+        actual_version = importlib.import_module(module_name).__version__
+        min_version, max_version, _ = _get_min_max_version_and_pip_release(module_key)
+        return _check_version_in_range(actual_version, min_version, max_version)
+
+    return all(
+        [
+            _check_supported(module_name, module_key)
+            for module_name, module_key in _cross_tested_flavor_to_module_name_and_module_key[
+                flavor_name
+            ]
+        ]
+    )
+
+
+def gen_autologging_package_version_requirements_doc(flavor_name):
+    def _gen_single_requirement(module_key):
+        min_ver, max_ver, pip_release = _get_min_max_version_and_pip_release(module_key)
+        return "``{min_ver}`` <= ``{pip_release}`` <= ``{max_ver}``".format(
+            min_ver=min_ver, pip_release=pip_release, max_ver=max_ver
+        )
+
+    required_pkg_versions = ",".join(
+        [
+            _gen_single_requirement(module_key)
+            for _, module_key in _cross_tested_flavor_to_module_name_and_module_key[flavor_name]
+        ]
+    )
```

Comment:
In accordance with my comment below about us not needing to version check both `torch` and `pytorch-lightning` (`pytorch-lightning` is sufficient), I think we can refactor this so we only return a docstring for a single requirement.